docs/howtos/redisbloom/with-dotnet/index.md

```diff
@@ -0,0 +1,175 @@
+---
+id: redisbloom-withdotnet
+title: Using Redis Bloom with .NET
+sidebar_label: Using Redis Bloom with .NET
+slug: /howtos/redisbloom/with-dotnet/redisbloom-withdotnet
+authors: [steve]
+---
+
+Using RedisBloom allows you to efficiently keep track of presence, heavy hitters, and counts on large streams of data. To use RedisBloom in .NET, you should use the [StackExchange.Redis](https://github.com/stackexchange/stackexchange.redis) library. To get started with that package, follow our [getting started guide](https://developer.redis.com/develop/dotnet/). Once you have a reference to an `IDatabase` object, you will need to use the `db.Execute` and `db.ExecuteAsync` methods to run the custom commands you want against redis bloom.
+
+## Bloom Filters
+
+Bloom Filters are a powerful data structure that can tell if an item is in a set, think a username on a sign-up form. They're incredibly compact, requiring only 10-20 bits per item you want to add, and extremely quick to add items to, and equally fast to determine if an item is in a set or not. 
+
+### Create a Filter
+
+You don't need to create a Bloom Filter explicitly as any call of `BF.ADD` to a non-existent key will automatically create a Bloom Filter for you. However, if you want to tell Redis ahead of time how much data the Bloom Filter can expect and the error rate that you want for that data (the number of false positives it will report), You can use the `BF.RESERVE` command:
+
+```csharp
+await db.ExecuteAsync("BF.RESERVE", "bf:username", .01, 10000);
+```
+
+The above command will reserve a Bloom Filter on the key `bf:username` that expects 10000 records and will have an error rate of 1%.
+
+### Adding to a Filter
+
+To add to a Bloom Filter, all you need is to use the `BF.ADD` command:
+
+```csharp
+await db.ExecuteAsync("BF.ADD", "bf:username", "Kermit");
+```
+
+The preceding code will add the username `Kermit` to the `bf:username` filter.
+
+### Check if an Item is in a Filter
+
+To check if an item has been added to a bloom filter yet, you will use the `BF.EXISTS` command:
+
+```csharp
+var exists = await db.ExecuteAsync("BF.EXISTS", "bf:username", "Kermit") == 1; 
+```
+
+After running that command, if the Bloom Filter reports that it contains the item, `exists` will be true; otherwise, `exists` will be false.
+
+## Count-Min Sketch
+
+You can use Count-Min Sketches to count the number of times an item has been added to a set quickly and compactly. Although, of course, like other probabilistic data structures, it has some probability of error. In this case, it can over count the number of occurrences. The dimensions of the sketch determine the likelihood of this.
```

Comment:
```suggestion
You can use Count-Min Sketches to count the number of times an item has been added to a set quickly and compactly. Although, of course, like other probabilistic data structures, it has some margin of error. In this case, it can over count the number of occurrences. The dimensions of the sketch determine the likelihood of this.
```